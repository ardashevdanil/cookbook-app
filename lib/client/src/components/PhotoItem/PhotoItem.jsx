import PropTypes from 'prop-types';
import React from 'react';
import './PhotoItem.css';

function PhotoItem({ content }) {
  return (
    <div className="PhotoItem">
      <div className="PhotoItem__img">
        <img
          alt={content.name}
          src={content.img}
        />
      </div>
    </div>
  );
}

PhotoItem.propTypes = {
  content: PropTypes.shape({
    img: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
  }).isRequired,
};

export default PhotoItem;
