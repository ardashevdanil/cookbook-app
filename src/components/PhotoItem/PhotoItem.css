.PhotoItem {
	width: 312px;
	padding: 20px calc( (100% - 312px * 3)/6 );
}

.PhotoItem__img {
  height: 190px;
  overflow: hidden;
}

.PhotoItem img {
	width: 100%;
}

@media only screen and (max-width: 1120px) {
  .PhotoItem {
    width: 100%;
    padding-top: 0;
  }

  .PhotoItem>img {
    height: 64.7vw;
  }
}