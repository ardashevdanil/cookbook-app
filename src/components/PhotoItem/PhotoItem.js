import PropTypes from 'prop-types';
import React from 'react';
import './PhotoItem.css';

function PhotoItem(props) {

	return (
		<div className='PhotoItem'>
			<div className ='PhotoItem__img'>
				<img
					alt={ props.content.name}
					src={ props.content.img }
				/>
			</div>
		</div>
	);
}

PhotoItem.propTypes = {
	content: PropTypes.shape({
		img: PropTypes.string.isRequired,
		name: PropTypes.string.isRequired,
	})
}

export default PhotoItem;