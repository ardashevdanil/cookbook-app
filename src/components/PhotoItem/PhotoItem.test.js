import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

import PhotoItem from './PhotoItem';

Enzyme.configure({ adapter: new Adapter() });

describe('test PhotoItem component', () => {
  const pushCart = jest.fn();
	const component = Enzyme.shallow(
		<PhotoItem
      content={{
        img: '',
        name: 'test',
      }}
    />
	);

	test('renders correctly', () => {
		expect(component.debug()).toMatchSnapshot();
	});

  /*
  test('calls pushCart after click on the cart button', () => {
    component.find('div').at(6).simulate('click');

    expect(pushCart.mock.calls.length).toBe(1);
  })
  */
})