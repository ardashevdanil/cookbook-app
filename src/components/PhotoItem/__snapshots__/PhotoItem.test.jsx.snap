// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test PhotoItem component renders correctly 1`] = `
"<div className=\\"PhotoItem\\">
  <div className=\\"PhotoItem__img\\">
    <img alt=\\"test\\" src=\\"\\" />
  </div>
</div>"
`;
